### PR TITLE
Fix: issues with drag and drop in Chrome 96

### LINKED ISSUE #6254: [BUG] importing by drag-and-drop of tiddler title is broken in latest update to Chrome and Edge
As reported here: https://talk.tiddlywiki.org/t/drag-and-drop-does-not-work-in-chrome-96-0/1605

1. open https://tiddlywiki.com/ in your browser
2. create a tiddler named "TestTiddler" with text content "this is a test tiddler"
3. open https://tiddlywiki.com/empty in another browser tab
3. drag TestTiddler from https://tiddlywiki.com/ and drop into the empty.html
5. The $:/Import shows "download.url", not "TestTiddler"
6. On import "download.url" tiddler is created with a text field containing:
`[InternetShortcut] URL=data:text/vnd.tiddler,%7B%22created%22%3A%2220211120075141902%22%2C%22text%22%3A%22this%20is%20a%20test%20tiddler%22%2C%22tags%22%3A%22%22%2C%22title%22%3A%22TestTiddler%22%2C%22modified%22%3A%2220211120075157966%22%7D`

**TiddlyWiki Configuration (please complete the following information):**
 - v5.2.0
 
**Desktop (please complete the following information):**
 - Windows10
 - Chrome 96.0.4664.45 (Official Build) (64-bit) or Edge 96.0.1054.29 (Official build) (64-bit)

### PR DESCRIPTION
**Pull request by saqimtiaz**
Fixes issues with drag and drop (false positives in detection for files being dropped) introduced by Chrome 96. Fixes #6254

* in $dropzone, do not attempt to import files if the `dataTransfer` property of the event contains the type `text/vnd.tiddler`
* in the editor, do not attempt to import files if the `dataTransfer` property of the drop event contains types other than `Files`. Testing for the type `text/plain` is consistent across browsers, this type is never present if files are being dropped.

Tested with latest Chrome and Firefox on Ubuntu and Windows 10, and latest Chrome and Safari on OSX.

### CODE REVIEW

File: core/modules/utils/dom/dragndrop.js [hunk 1]
```diff
@@ -220,4 +220,10 @@ exports.dragEventContainsFiles = function(event) {
 	return false;
 };
 
+exports.dragEventContainsFiles = function(event) {
+	return (dragEventContainsType(event,"Files") && !$tw.utils.dragEventContainsType(event,"text/plain"));
```
Comment (Jermolene) [inline core/modules/utils/dom/dragndrop.js:224]: I was confused by the two different ways of addressing the dragEventContainsType function. Could it be like this:

```
return (dragEventContainsType(event,"Files") && !dragEventContainsType(event,"text/plain"));
```

**Comment by Jermolene** [comment]
Great stuff thanks @saqimtiaz

File: core/modules/utils/dom/dragndrop.js (hunk 1 above)
Comment (saqimtiaz) [reply core/modules/utils/dom/dragndrop.js:224]: @Jermolene you are absolutely right, that is an artefact from a late refactoring. Will resolve now.

**Comment by saqimtiaz** [comment]
@Jermolene I have resolved that inconsistency, good spot, thank you.

**Comment by Jermolene** [comment]
Great thanks @saqimtiaz

**Comment by AnthonyMuscio** [comment]
@Jermolene would it be difficult to drop a fix on 5.2.0 for this?

**Pull request merged**